Add Platforms Tags and Page (#422)
typelevel/typelevel.github.com: README.md

That's it, we'll take care of the rest. If you wish, you can also submit just a plain Markdown file and we'll be happy to integrate it.

> You can also use `tut` in posts. See `posts/2016-09-30-subtype-typeclasses.md` for an example.

Reviewer: `tut` is archived, and we don't have any mdoc examples. We can investigate that as a follow up though :)

### Adding a project

There are three types of projects: organization projects, affiliate projects, and core/featured projects.

To add an organization project, insert a new entry, alphabetically, in the `_data/projects.yml` file with the following keys:

```yml
- title: "Cats"

>   description: "An experimental library intended to provide abstractions for functional programming in Scala, leveraging its unique features. Design goals are approachability, modularity, documentation and efficiency."

Reviewer 2: ```suggestion
  description: "An experimental library that provides abstractions for functional programming in Scala, leveraging its unique features. Design goals are approachability, modularity, and efficiency."
```
optional... I would think "documentation" is sort of a minimum implicit requirement for all projects here and sounds odd to call it a "design goal", though  I acknowledge Cats' docs are among the best--and that this diff in your PR is sorta incidental.

Author: This feedback is good, however, this isn't actually the place to change the description of `cats` that would be in the `_data/projects.yml` file, and I'd like to avoid yak shaving that here as well.

So I've just updated this README snippet to match the current content in the data file.

  github: "https://github.com/non/cats"
  platforms: [js, jvm, native]
  permalink: "https://non.github.io/cats" # optional
```

Right now nothing more than the correct front matter is required.

> To add an:
> - **affiliate** project, add `affiliate: true` to the project entry
> - **core** project, add `core: true` to the project entry

Reviewer 2: ```suggestion
To add
- an **affiliate** project, add `affiliate: true` to the project entry
- a **core** project, add `core: true` to the project entry
```

### Adding a page

> To add a page, create a directory in the root of the project, with at least an `index.html` file in that directory. The site navigation is not fully dynamic for simplification purposes. The new page can be added to the nav by updating `_data/nav.yml`.

Reviewer 2: ```suggestion
To add a page, 

1. Create a directory in the root of the project, with at least an `index.html` file in that directory. 2. Update  `_data/nav.yml` to add it to the navigation. (The site navigation is not fully dynamic for simplification.)
```

### Adding a project

There are three types of projects: organization projects, affiliate projects, and core/featured projects.

To add an organization project, insert a new entry, alphabetically, in the `_data/projects.yml` file with the following keys:

```yml
- title: "Cats"
  description: "A library intended to provide abstractions for functional programming in Scala, leveraging its unique features. Design goals are approachability, modularity, documentation and efficiency."
  github: "https://github.com/typelevel/cats"
  platforms: [js, jvm, native]
  permalink: "https://typelevel.org/cats/" # optional
```

Right now nothing more than the correct front matter is required.

To add
- an **affiliate** project, add `affiliate: true` to the project entry
- a **core** project, add `core: true` to the project entry

### Adding a page

To add a page, 

> 1. Create a directory in the root of the project, with at least an `index.html` file in that directory. 2. Update  `_data/nav.yml` to add it to the navigation. (The site navigation is not fully dynamic for simplification.)

Reviewer: ```suggestion
1. Create a directory in the root of the project, with at least an `index.html` file in that directory. 
2. Update  `_data/nav.yml` to add it to the navigation. (The site navigation is not fully dynamic for simplification.)
```

```yml
layout: post
title: "Cats"

> category: "Functional Programming"

Reviewer: Add the platforms line as part of the example?

I can't add it as a suggestion :( 
`platforms: [js, jvm, native]`

Reviewer: Oh closer inspection, this whole section might need a bit of a revamp. Line 128 is still referring to incubator projects, and the example also includes a `layout` key that I don't think is necessary.

Reviewer: I updated the readme in [54fcf5c](https://github.com/typelevel/typelevel.github.com/pull/422/commits/54fcf5cd2f2bf92ed89663652221ca9507a6f78c)
